### PR TITLE
Feat(eos_designs): Support OSPF routing in tenant VRFs

### LINKED ISSUE #1400: Add Tenant VRF OSPF support in eos_designs
### Enhancement summary

Implement OSPF data model eos_designs tenant vrfs and relevant ospf commands for l3_interface and svis.

### Which component of AVD is impacted

eos_designs

### Use case example

Use case is having tenant OSPF routing on l3_interfaces and SVIs.

### Describe the solution you would like

```yaml
tenants:
  TENANT_A:
    vrfs:
      TEN_A_VRF1:
        ospf:
          < ... >
    svis:
      1:
        ospf:
          < ... >
    l3_interfaces:
     - interfaces: [ ... ]
        ospf:
          < ... >
```

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by emilarista**
## Change Summary

This PR adds tenant ospf configuration data model to eos_designs and extends l3_interface and svi data models to support ospf commands.

## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

This data model for tenant ospf configuration, l3_interfaces and svi settings:

```yaml
tenants:
  < tenant_a >:
    vrfs:
      < tenant_a_vrf_1 >:
        ...
        ospf:
          enabled: < true | false >
          process_id: < int, Default -> vrf_vni
          router_id: < router-id, Default -> switch router_id >
          max_lsa: < int >
          bfd: < true | false, Default -> false >
          redistribute_bgp:
            enabled: < true | false >
            route_map: < route-map name >
          nodes:
            - < hostname1 >
            - < hostname2 >
        svis:
          < 1-4096 >:
            ...
            ospf:
              enabled: < true | false >
              point_to_point: < true | false, Default -> false >
              area: < ospf area id, Default -> 0 >
              cost: < ospf link cost >
              authentication: < simple | message-digest >
              simple_auth_key: < password used with simple authentication >
              message_digest_keys:
                - id: < int >
                  hash_algorithm: < md5 | sha1 | sha256 | sha384 | sha512, Default -> sha512 >
                  key: < key password >
        l3_interfaces:
         - interfaces: [ <interface_name1>, <interface_name2>, <interface_name3> ]
            ...
            ospf:
              enabled: < true | false >
              point_to_point: < true | false, Default -> true >
              area: < ospf area id, Default -> 0 >
              cost: < ospf link cost >
              authentication: < simple | message-digest >
              simple_auth_key: < password used with simple authentication >
              message_digest_keys:
                - id: < int >
                  hash_algorithm: < md5 | sha1 | sha256 | sha384 | sha512, Default -> sha512 >
                  key: < key password >
```

## How to test

Added tests to molecule eos_designs_unit_tests and tested in lab environment.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md [hunk 1]
```diff
@@ -241,6 +274,17 @@ tenants:
               < multiline eos cli >
             # Custom structured config added under ethernet_interfaces.<interface> for eos_cli_config_gen
             structured_config: < dictionary >
+            ospf:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md:282]: add one blank line above

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-bgp-vrfs.j2 [hunk 2]
```diff
@@ -75,9 +75,18 @@
 {%                             break %}
 {%                         endif %}
 {%                     endfor %}
-{%                 elif vrf.redistribute_static == true %}
+{%                 elif vrf.redistribute_static is arista.avd.defined(true)  %}
         - static
 {%                 endif %}
+{%             endif %}
+{%             if vrf.ospf is arista.avd.defined %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-bgp-vrfs.j2]: simplify this logic:
```jinja
{% if vrf.ospf.enabled is arista.avd.defined(true) and vrf.redistribute_ospf is not arista.avd.defined(false) %}
{%     if inventory_hostname in vrf.ospf.nodes | arista.avd.default([inventory_hostname]) %}
- ospf
{%     endif %}
{% endif %}
```
The current logic would configure redistribution if redistribute_ospf == true even if the node was not defined under `nodes`.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/main.j2 [hunk 3]
```diff
@@ -35,6 +35,8 @@
 
 {%     include 'network_services/eos-cli-vrfs.j2' %}
 
+{%     include 'network_services/router-ospf.j2' %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/main.j2:36]: This should be moved up above the `eos-cli-vrfs.j2`

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+{%         endif %}
+{%     endfor %}
+{% endfor %}
+{% if network_services_data.ospf_configured is arista.avd.defined(true) %}
+router_ospf:
+  process_ids:
+{%     for tenant in network_services_data.tenants %}
+{%         for vrf in tenant.vrfs %}
+{%             if vrf.ospf is arista.avd.defined %}
+{%                 if vrf.ospf.nodes is not arista.avd.defined or inventory_hostname in vrf.ospf.nodes %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: Use same logic as line 4 `if inventory_hostname in vrf.ospf.nodes | arista.avd.default([inventory_hostname])`. Same below for l3_interfaces

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+{%                     set vrf_ospf_interfaces = [] %}
+{%                     set ospf_process_id = vrf.ospf.process_id | arista.avd.default(vrf.vrf_vni, vrf.vrf_rt, vrf.vrf_id) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: IMO you should only default to `vrf_id`. vni could then _also_ default to `vrf_id`. There is no variable `vrf_rt`

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+{%                     set ospf_router_id = vrf.ospf.router_id | arista.avd.default(switch.router_id) %}
+{%                     for l3_interface in vrf.l3_interfaces | arista.avd.natural_sort %}
+{%                         if l3_interface.nodes is arista.avd.defined and inventory_hostname in l3_interface.nodes and l3_interface.interfaces is arista.avd.defined %}
+{%                             for node in l3_interface.nodes %}
+{%                                 if node == inventory_hostname and l3_interface.ospf.enabled is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: do the check for `ospf.enabled` first - in the same line where you check if `inventory_hostname` is in `nodes`

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+{%                                     do vrf_ospf_interfaces.append(l3_interface.interfaces[loop.index0]) %}
+{%                                 endif %}
+{%                             endfor %}
+{%                         endif %}
+{%                     endfor %}
+{%                     for svi in vrf.svis | arista.avd.natural_sort %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: no need to sort SVI any longer

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+{%                         if svi.ospf.enabled is arista.avd.defined(true) %}
+{%                             do vrf_ospf_interfaces.append("Vlan" ~ svi.id) %}
+{%                         endif %}
+{%                     endfor %}
+    {{ ospf_process_id }}:
+      vrf: {{ vrf.name }}
+      passive_interface_default: true
+      router_id: {{ ospf_router_id }}
+      no_passive_interfaces:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: could be simplified to just `no_passive_interfaces: {{ vrf_ospf_interfaces | arista.avd.natural_sort }}

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        - {{ interface }}
+{%                     endfor %}
+{%                     if vrf.ospf.bfd is arista.avd.defined(true) %}
+      bfd_enable: true
+{%                     endif %}
+{%                     if vrf.ospf.max_lsa is arista.avd.defined() %}
+      max_lsa: {{ vrf.ospf.max_lsa }}
+{%                     endif %}
+      redistribute:
+{%                     if vrf.ospf.redistribute_bgp.enabled is not arista.avd.defined or vrf.ospf.redistribute_bgp.enabled is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: could be rewritten to `if vrf.ospf.redistribute_bgp.enabled | arista.avd.default(true) == true`

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2 [hunk 10]
```diff
@@ -75,6 +75,25 @@ vlan_interfaces:
         vrf: {{ svi_ip_helpers[helper_ip].source_vrf | arista.avd.default }}
 {%                 endfor %}
 {%             endif %}
+{%             if svi.ospf.enabled is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2]: add a check if `vrf.ospf.enabled is arista.avd.defined(true)`

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2 [hunk 11]
```diff
@@ -75,6 +75,25 @@ vlan_interfaces:
...
+    ospf_area: {{ svi.ospf.area | arista.avd.default(0) }}
+    ospf_network_point_to_point: {{ svi.ospf.point_to_point | arista.avd.default(false) }}
+{%             endif %}
+{%             if svi.ospf.cost is arista.avd.defined() %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2]: remove empty `()` (same elsewhere) - i know it is inconsistent everywhere, but let's try to remove it with we build new stuff.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+{%             set network_services_data.ospf_configured = true %}
+{%         endif %}
+{%     endfor %}
+{% endfor %}
+{% if network_services_data.ospf_configured is arista.avd.defined(true) %}
+router_ospf:
+  process_ids:
+{%     for tenant in network_services_data.tenants %}
+{%         for vrf in tenant.vrfs %}
+{%             if vrf.ospf is arista.avd.defined %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: change to `vrf.ospf.enabled is arista.avd.defined(true)`

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/ethernet-interfaces.j2 [hunk 13]
```diff
@@ -33,6 +33,25 @@ ethernet_interfaces:
 {%                         if l3_interface.description is arista.avd.defined %}
     description: {{ l3_interface.description }}
 {%                         endif %}
+{%                         if l3_interface.ospf.enabled is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/ethernet-interfaces.j2]: add a check if `vrf.ospf.enabled is arista.avd.defined(true)`

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/ethernet-interfaces.j2 (hunk 13 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/ethernet-interfaces.j2]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 (hunk 12 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2 (hunk 11 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2]: Fixed in all network_services templates.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2 (hunk 10 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 (hunk 9 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md (hunk 1 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md:282]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/main.j2 (hunk 3 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/main.j2:36]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-bgp-vrfs.j2 (hunk 2 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-bgp-vrfs.j2]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 (hunk 4 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: Fixed. Had to add | arista.avd.natural_sort to nodes below l3_interfaces to catch l3_interfaces with undefined nodes.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 (hunk 5 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: Added new variable vrf_id, defaulted vrf_vni to vrf_if for services and defaulted ospf.process_id to vrf_id instead of vrf_vni.

File: ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md [hunk 14]
```diff
@@ -106,6 +106,11 @@ tenants:
         # If vrf_vni > 10000 make sure to adjust mac_vrf_vni_base accordingly to avoid overlap.
         vrf_vni: < 1-1024 >
 
+        # VRF ID | Required.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md]: I don't think we need to specify this as required. I would suggest rewording:
```yaml
# VRF ID | Optional (required if "vrf_vni" is not set)
# vrf_id is used as default value for "vrf_vni" and "ospf.process_id" unless those are set.
vrf_id: < 1-1024 >

File: ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md [hunk 15]
```diff
@@ -106,6 +106,11 @@ tenants:
         # If vrf_vni > 10000 make sure to adjust mac_vrf_vni_base accordingly to avoid overlap.
         vrf_vni: < 1-1024 >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md]: `vrf_vni: < 1-1024 | default -> vrf_id >`

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2 [hunk 16]
```diff
@@ -75,6 +75,25 @@ vlan_interfaces:
         vrf: {{ svi_ip_helpers[helper_ip].source_vrf | arista.avd.default }}
 {%                 endfor %}
 {%             endif %}
+{%             if svi.ospf.enabled is arista.avd.defined(true) and vrf.ospf.enabled is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2:78]: I think you want this `if` as a condition for all the OSPF knobs below. Not only area and point-to-point.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2 [hunk 17]
```diff
@@ -75,6 +75,25 @@ vlan_interfaces:
...
+    ospf_cost: {{ svi.ospf.cost }}
+{%             endif %}
+{%             if svi.ospf.authentication is arista.avd.defined("simple") and svi.ospf.simple_auth_key is arista.avd.defined %}
+    ospf_authentication: "simple"
+    ospf_authentication_key: {{ svi.ospf.simple_auth_key }}
+{%             elif svi.ospf.authentication is arista.avd.defined("message-digest") and svi.ospf.message_digest_keys is arista.avd.defined %}
+    ospf_authentication: "message-digest"
+    ospf_message_digest_keys:
+{%                 for key in svi.ospf.message_digest_keys | arista.avd.natural_sort %}
+      {{ key.id }}:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2:92]: no variable protection for `key.id` and `key.key`

File: ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md (hunk 14 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md (hunk 15 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2 (hunk 16 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2:78]: Yep, thanks, applied this to vlan-interfaces and ethernet-interfaces as well.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2 (hunk 17 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2:92]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 (hunk 8 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: this would also ensure that we don't create an empty key, but instead an empty list.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 (hunk 6 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 (hunk 7 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: Fixed.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 (hunk 8 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: Implemented the suggested fix.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 [hunk 18]
```diff
@@ -0,0 +1,54 @@
+{# Rendering tenant OSPF #}
+{% for tenant in network_services_data.tenants %}
+{%     for vrf in tenant.vrfs %}
+{%         if vrf.ospf is arista.avd.defined and inventory_hostname in vrf.ospf.nodes | arista.avd.default([inventory_hostname]) %}
+{%             set network_services_data.ospf_configured = true %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2:5]: You can set a break after ```ospf_configured``` is set to true, because it has to be true only once. This will save some time if there are 1000's of VRFs. 

Or you can use a different algorithm to save those tenants and respective VRFs in a temporary variable if ```vrf.ospf.enabled is true``` and the other check is also true.
Now you can loop through the temp variable like you did in the line 9 to write the configuration.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 [hunk 19]
```diff
@@ -0,0 +1,54 @@
+{# Rendering tenant OSPF #}
+{% for tenant in network_services_data.tenants %}
+{%     for vrf in tenant.vrfs %}
+{%         if vrf.ospf is arista.avd.defined and inventory_hostname in vrf.ospf.nodes | arista.avd.default([inventory_hostname]) %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: Check for ```vrf.ospf.enabled is arista.avd.defined(true)```, just suggestion but not mandatory.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+      router_id: {{ ospf_router_id }}
+      no_passive_interfaces: {{ vrf_ospf_interfaces | arista.avd.natural_sort }}
+{%                 if vrf.ospf.bfd is arista.avd.defined(true) %}
+      bfd_enable: true
+{%                 endif %}
+{%                 if vrf.ospf.max_lsa is arista.avd.defined %}
+      max_lsa: {{ vrf.ospf.max_lsa }}
+{%                 endif %}
+      redistribute:
+{%                 if vrf.ospf.redistribute_bgp.enabled | arista.avd.default(true) == true %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: We don't need to check if it's ```== true``` for a boolean. You can modify this line 44 to ```if vrf.ospf.redistribute_bgp.enabled | arista.avd.default(true)``` 

If the value is defined false then the logic would be ``` if false``` or else ``` if true```. 

FYI, the article on python: https://switowski.com/blog/checking-for-true-or-false

File: ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md [hunk 21]
```diff
@@ -138,6 +142,26 @@ tenants:
...
+        # created on all leafs where the vrf is deployed. This will also cause automatic ospf redistribution into bgp unless
+        # explicitly turned off with "redistribute_ospf: false".
+        ospf:
+          enabled: < true | false >
+          process_id: < int, Default -> vrf_id >
+          router_id: < router-id, Default -> switch router_id >
+          max_lsa: < int >
+          bfd: < true | false, Default -> false >
+          redistribute_bgp:
+            enabled: < true | false >
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md]: You have set this to ```true```, default value. So please document that as well.

File: ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md [hunk 22]
```diff
@@ -138,6 +142,26 @@ tenants:
...
+            route_map: < route-map name >
+          nodes:
+            - < hostname1 >
+            - < hostname2 >
+
+        # Non-selectively enabling or disabling redistribute ospf inside the VRF | Optional.
+        redistribute_ospf: < true | false >
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md]: The logic described in J2 template suggests this value is set to True by default. Because you are checking only ```not false``` condition. So please document this default value.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 (hunk 18 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2:5]: Implemented the break suggestion.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 (hunk 19 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: Good idea! Have implemented this.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2 (hunk 20 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2]: Implemented the suggested fix.

File: ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md (hunk 21 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md]: Updated documentation accordingly.

File: ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md (hunk 22 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/doc/network-services.md]: Updated documentation accordingly.

File: ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2 [hunk 23]
```diff
@@ -75,6 +75,28 @@ vlan_interfaces:
         vrf: {{ svi_ip_helpers[helper_ip].source_vrf | arista.avd.default }}
 {%                 endfor %}
 {%             endif %}
+{%             if svi.ospf.enabled is arista.avd.defined(true) and vrf.ospf.enabled is arista.avd.defined(true) %}
+    ospf_area: {{ svi.ospf.area | arista.avd.default(0) }}
+    ospf_network_point_to_point: {{ svi.ospf.point_to_point | arista.avd.default(false) }}
+
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2]: Is this blank line required? If not please remove this

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2]: Apart from this LGTM!

Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/network_services/vlan-interfaces.j2]: Good catch, removed this!

**Review by tgodaA — APPROVED** [review]
LGTM!

**Review by ClausHolbechArista — APPROVED** [review]
LGTM
